grid/deploy/component.py

```diff
@@ -0,0 +1,106 @@
+import sys
+import subprocess
+import os
+from abc import ABC
+
+from grid import utils as gr_utils
+
+
+class BaseDeployment(ABC):
+    """ Abstract Class used to instantiate generic attributes for other deployment classes. """
+
+    def __init__(self, env_vars, verbose: bool = True):
+        """
+        Args:
+            env_vars: Environment vars used to configure component
+            verbose: Used to define level of verbosity
+            logs: list of logs used in deploy process
+            commands: list of commands used in deploy process
+        """
+        self.env_vars = env_vars
+        self.verbose = verbose
+        self.logs = list()
+        self.commands = list()
+
+    def deploy_on_heroku(self):
+        """ Method used to deploy component on heroku platform. """
+        raise RuntimeError("Heroku deployment not specified!")
+
+    def deploy_on_container(self):
+        """ Method used to deploy component using docker containers. """
+        raise RuntimeError("Container deployment not specified!")
+
+    def _check_dependency(
+        self,
+        lib="git",
+        check="usage:",
+        error_msg="Error: please install git.",
+        verbose=False,
+    ):
+        """ This method checks if the environment have a specific dependency.
+            Args:
+                dependency_lib : libs that will be verified.
+                check: specific string to check if app was installed.
+                error_msg: If not installed, raise an Exception with this.
+                verbose: Used to define level of verbosity.
+            Raises:
+                Exception: if not installed, raise a standard Exception
```

Comment:
Could we raise a more specific exception? Maybe a runtime exception is better.

Reply:
Done! (b2843ea)